### DISCUSSION TITLE
Did I pick the wrong starter theme?

### DISCUSSION THREAD

**Discussion by goclimbing** [bootscore/bootscore#886] [General]
A couple years ago I found the Understrap starter theme and have been using it since.  I just now found this starter theme. It looks like it has better support and a more active community.  Anyone out there able to give me a basic comparison or your input on the community and support?

I'm guessing from the name that like with Understrap bootscore is based on the underscores _s theme?

I'll look under the hood but I suspect bootscore and understrap are very similar in terms of the devops.

Looking forward to learning about this theme.

Thanks!

**Comment by goclimbing** [comment]
You guys are killing me. This looks great!  Problem is I'm 8 months into developing a huge project based in Understrap.  The upside is if i decide to switch I'll get paid for it.

**Reply by crftwrk** [reply]
Yes, Bootscore is based on the _s theme as well and is similar to the Understrap theme. The key differences are:

- built-in scssphp compiler. No need to install Gulp or anything else, just add your styles to the scss files.
- Full WooCommerce support (if installed)
- Support for the GB editor styles and block patterns
- Always updated to the latest libraries
- Custom plugins to extend the theme (free and paid)
- and much more...

> The upside is if i decide to switch I'll get paid for it.

Do it ;-)

**Reply by hsankala** [reply]
I completely agree with what Basti mentioned about this theme. I started with a simple website and notes, as I’ve always loved the Bootstrap framework, especially Bootstrap 5. I wanted a WordPress theme that was very Bootstrap 5-centric. Initially, I used the UnderStrap theme, which is a decent choice. It integrates Bootstrap 5 reasonably well and puts it front and center.

However, I eventually switched to BootScore because of its fantastic WooCommerce integration and the excellent work by its developers. BootScore has a reliable release schedule with updates throughout the year, whereas UnderStrap's updates were more sporadic. BootScore just felt more polished and well-maintained, with consistent attention from its lead developers over the years.

It also offers some really good developer-friendly features. As Basti mentioned, I initially used BootScore with a Gulp compiler, but I found that setup a bit overkill for the project I was working on. I’ve since switched to the built-in SCSS PHP compiler, which works perfectly for my needs.

One of the standout features for me, which Basti also highlighted, is that BootScore is always updated to include the latest libraries. If you're curious, you can see my e-commerce site built with it: [churchantiques.com](https://churchantiques.com/).

**Comment by goclimbing** [comment]
It's gonna be a long weekend migrating. But just got out of our weekly IT meeting and I got the full sign off to do it.  They play the long game.

I'm also suspecting there's better Bootstrap integration like 'cards' in grid display, etc. My company is a big deal in the metals and bullion market and has been for decades.

The GB integration will probably also be helpful as I have custom ACF<->GB blocks built.

Also noticed the Dark plugin and was looking at understrap trying to figure out how to integrate that.  Now I don't have too...

**Comment by goclimbing** [comment]
What's the method for building the theme (locally) for use on the site being developed?  I use an Action from git to push to stage for testing but build the theme for the live site?  Zip it and upload it?

**Reply by hsankala** [reply]
I'd take a peek at the documentation for this, and as far as I can tell, it's all driven by a PHP-powered SCSS compiler. You can find the documentation here: https://bootscore.me/documentation/theme/css-scss-compiler/.

From what I understand, the compiler watches the SCSS dicrectory/files for changes. When it detects any updates, it automatically compiles the changes in the background. I find it fast — if I make a change, I only need to wait a couple of seconds before the CSS is updated for the theme.

This setup is convenient for the production server too. If you need to make a quick fix to the SCSS, you can tweak it, and the PHP SCSS compiler will handle the compilation directly on the server. So it works the same way locally as well as remotely, which is a nice consistency.

One thing it doesn't do, however, is minification or CSS and JS bundling, as far as I'm aware. When I moved over from GULP tasks to fully embrace the PHP SCSS compiler that is bundled with the theme, I started using a caching plugin to ensure that the CSS and JS are both minified and bundled efficiently.

**Reply by goclimbing** [reply]
Thanks for the tips. I've been spending the morning reading the docs. There's a lot there.  I was about to ask how to change the logo but a quick search of the forum netted the results i needed.  Guess I'll be even better with hooks.

So far I have my custom css and css variables moved over and working.  ACF is up and running with my GB blocks.

But here's one...

The site I'm building gets our product catalog via api from our metals vendor.  We originally only got the product title, metal type and price so I used OpenAI to populate the short and long descriptions, and SEO tag. Eventually they added images that update via api to our catalog.

The price constantly changes and is updated via the vendors plugin into our Woo DB.  So I wrote an ajax script to check for a new price every 2 minutes and then update via jquery on the single item page (will also get implemented on cart and probably checkout).  

Problem is if you look close at the price HTML it doesn't have a "clean" wrapper to target so I added a div around the price class="product_price" which I can target.  The woo guys know about this and asked me to join the developers forum as they felt this was/is a unique case use and may add it.  Is there a way with hooks or filters to wrap the price in the div i need?

**Reply by hsankala** [reply]
I'm not sure if I fully grasp the dynamic price updating solution you're using, especially with pulling data from an external wholesaler or vendor’s API. However, it does seem like you could directly update the price in the WooCommerce database instead of relying on jQuery and client-side manipulation to reflect the change on the page.

Another solution: When the API detects a new price that’s different from what’s listed in the product, you can directly update the price in the WooCommerce database using server-side methods. This would ensure that the price update is reflected everywhere — on the product page, in the cart, and during checkout — without needing to update it with client-side scripting.

I could be misunderstanding your approach though, and there might be a specific reason you're using jQuery for this.

**Reply by goclimbing** [reply]
It's possible that the price is updated in the Woo DB via the API after the page renders (it hardly ever does but it was a request from the higher ups).    All the pricing updates are handled by the vendors plugin so we have no control nor want it.  I just need to make sure the viewing customer has the most recent price after page load for that $80k brick of gold they are about to purchase.

**Reply by hsankala** [reply]
I understand where you're coming from — when big money is involved, even small price fluctuations (like 1% or 2%) can make a significant impact and potentially put customers off if they aren't properly informed.   May I ask the website your working on out of interest. 

Regarding your jQuery script, it seems functional for your use case, but the main developers regularly maintaining this project might be better positioned to provide a definitive answer.

From my experience with the Bootscore theme, it receives frequent updates for WooCommerce Template Updates and the devs are keep Woocommerce templates well sync'ed with WooCommerce updates as the folks at Automattic -WooCommerce often update these WooCommerce templates. 

The Bootscore team does a great job of integrating these Templates updates, essentially "bootstrapifying" them with lots of Bootstrap class enhancements. However, I have some doubt they would make significant structural changes to IDs or key elements in the templates. The reason I say this is: they aware of the vast WordPress ecosystem relies on these structures, and altering them too much could break compatibility with third-party plugins and customizations. Cosmetic changes, like adding Bootstrap classes, are one thing, but altering core identifiers is another matter entirely IMO.

As such, I don’t believe the Bootscore Woocommerce templates would make it an easier or more difficult for that matter to target specific page elements like the price element directly, apart from the Bootstrap enhancements. Still, the main maintainers of this repository might reply to this  and to provide more clarity.

In the meantime, I’d suggest I'd look have a nice an CSS specific rule combinator (likely your doing this already) that isolates the WooCommerce price element on the product page. This way, you can ensure it only applies to the product price dynamically displayed on the product page (assuming you don’t need it for category pages or elsewhere).

**Reply by goclimbing** [reply]
I can't disclose the client/site just yet.  It's been 8 months in dev and just yesterday we presented it to the top dog and he loved our design implementation and overall functionality. I want nothing more than to launch and share it with the world.

I hear what you're getting at but the problem was I couldn't find a good css rule class combination to hit that specific content.  So I copied price.php over into child and added a wrapper.

(It's not letting me copy my html in as code. weird)
![Screenshot 2024-12-06 at 2 27 33 PM](https://github.com/user-attachments/assets/b81e2c61-eaef-4683-ab6c-2c112c2de928)


I added the id="product-price" though now that I think about it I could probably add the currency symbol in the JS along the other tages and target woocommerce-Price-amount amount" which is probably what I'll do. 

The reason I asked if it could be done with hooks is because I also did the same thing to other templates where I needed the bootstrap classes (and didn't want to mess with hooks - just needed it done [for now] - and wasn't gonna mess with updating the woo css to my styles).  Looks like with this theme that was part of the purpose - bootstrap woo!

Pretty sure I can get rid of almost all custom woo templates I had to modify (adding btn btn-secondary text-white, etc). I'll just need to figure out how to tweak some hooks as this theme seems to be using btn-primary and I'm not wanting to swap the colors).

Also another  reason I jumped straight into this is I had to copy woo templates that were out of date into my child theme.  That's how often those guys update.

**Reply by hsankala** [reply]
Absolutely, that was one of the main reasons I was drawn to the Bootscore theme—it cleanly integrates Bootstrap with WooCommerce, making everything feel cohesive and streamlined. It’s clear the developers are passionate about the project, and the level of care they’ve put into it really shines through.

It’s a nice win to eliminate as many custom WooCommerce templates as possible. WooCommerce updates frequently, and having too many custom overrides just creates unnecessary maintenance headaches in the long run. It’s already challenging enough for the core maintainers Bootscore to keep things up to date, so minimizing that load for yourself is definitely the way to go.

Be interested once the site is out there to take a look, you can  share it in the thread where people display their Bootscore-themed projects—it would be great to see there when you finished: https://github.com/orgs/bootscore/discussions/317

Regarding hooks or filters for wrapping. One approach I’ve found might be a programmatic injection of your chosen ID is using a custom function, either in functions.php or within a custom plugin, to dynamically add unique ID to specific elements like prices. This approach keeps things clean, avoids template overrides, and makes targeting with JavaScript (or jQuery) straightforward.

For example, I asked ChatGPT generate this snippet for wrapping price elements dynamically:

```
function add_id_to_price_on_single_product( $price_html, $product ) {
    // Check if we're on a single product page
    if ( is_product() ) {
        // Get the current product ID
        $product_id = $product->get_id();
        // Create a unique ID based on the product ID
        $unique_id = 'hook-product-price-id-dynamic'
        // Wrap the price in a span with the unique ID
        $price_html = '<span id="' . esc_attr( $unique_id ) . '" class="custom-price-class">' . $price_html . '</span>';
    }
    return $price_html;
}
add_filter( 'woocommerce_get_price_html', 'add_dynamic_id_to_price_on_single_product', 10, 2 );
```

If you were able to get this approach to work it avoid directly modifying templates and ensures that the necessary flexibility is maintained.

**Reply by goclimbing** [reply]
Thanks. I'll try it. It looks like what I was looking for and at least eliminates that template.

And I will post there. I also have a personal site I'll migrate and will share as well.

**Reply by crftwrk** [reply]
@hsankala 

> One thing it doesn't do, however, is minification or CSS and JS bundling, as far as I'm aware

Default `main.css` is minified. If you enable the developer mode, then it`s beautyfied. Press the "Pretty print" button in dev console to show minified CSS:

<img width="309" alt="screen" src="https://github.com/user-attachments/assets/e7ed05c5-f4aa-483d-972f-8c9a9e42ac81">

The JavaScript uses the compiled `bootstrap-bundle.min.js`. No need to compile anything here.
